invenio_app_rdm/records_ui/views/records.py

```diff
@@ -55,41 +94,66 @@ def record_export(record=None, export_format=None, pid_value=None):
 
 
 @pass_record
-def record_file_preview(record=None, pid_value=None):
-    """Render a preview of."""
-    abort(404)
+@pass_file_metadata
+def record_file_preview(
+    record=None,
+    pid_value=None,
+    pid_type="recid",
+    file_metadata=None,
+    **kwargs
+):
+    """Render a preview of the specified file."""
+    if not file_metadata:
+        abort(404)
+
+    # Try to see if specific previewer is set
+    # TODO: what's the analog of: file_previewer = fileobj.get("previewer") ?
+    file_previewer = file_metadata.data.get("previewer")
+
+    # Find a suitable previewer
+    fileobj = PreviewFile(file_metadata, pid_value)
+    for plugin in current_previewer.iter_previewers(
+        previewers=[file_previewer] if file_previewer else None
+    ):
+        if plugin.can_preview(fileobj):
+            try:
+                return plugin.preview(fileobj)
+
+            except Exception:
+                current_app.logger.warning(
+                    (
+                        "Preview failed for {}, in {}:{}".format(
+                            fileobj.file.key,
+                            pid_type,
+                            pid_value,
+                        )
+                    ),
+                    exc_info=True,
+                )
```

Comment:
this just feels outdated, can probably be removed too?

Reply:
Yeah, back in the days it was because we had some super weird files that should be previewable but wasn't.